checks/kube/aws-irsa.sh

```diff
@@ -0,0 +1,1431 @@
+        SCRIPT_STATUS_OUTPUT=63
+        return
+    else
+        echo "[INFO] Found domain info: $domain_info"
+    fi
+
+    advanced_security_enabled=$(echo "$domain_info" | jq -r '.DomainStatus.AdvancedSecurityOptions.Enabled')
+    https_enforced=$(echo "$domain_info" | jq -r '.DomainStatus.DomainEndpointOptions.EnforceHTTPS')
+    domain_created=$(echo "$domain_info" | jq -r '.DomainStatus.Created')
+    domain_deleted=$(echo "$domain_info" | jq -r '.DomainStatus.Deleted')
+
+
+    if [[ "$advanced_security_enabled" != "true" ]]; then
+        echo "[FAIL] Advanced security is not enabled on OpenSearch domain $domain_name." 1>&2
+        SCRIPT_STATUS_OUTPUT=64
+    else
+        echo "[OK] Advanced security is enabled on OpenSearch domain $domain_name."
+    fi
+
+    if [[ "$https_enforced" != "true" ]]; then
+        echo "[FAIL] HTTPS is not enforced on OpenSearch domain $domain_name." 1>&2
+        SCRIPT_STATUS_OUTPUT=65
+    else
+        echo "[OK] HTTPS is enforced on OpenSearch domain $domain_name."
+    fi
+
+    if [[ "$domain_created" != "true" || "$domain_deleted" != "false" ]]; then
+        echo "[FAIL] OpenSearch domain $domain_name is either not created or marked as deleted." 1>&2
+        SCRIPT_STATUS_OUTPUT=66
+    else
+        echo "[OK] OpenSearch domain $domain_name is created and not marked as deleted."
+    fi
+
+    access_policy=$(echo "$domain_info" | jq -r '.DomainStatus.AccessPolicies')
+    echo "[INFO] Retrieved Access Policy for domain $domain_name: $access_policy"
+
+    # Check if the Access Policy allows access (basic check for "Effect": "Allow")
+    allow_access=$(echo "$access_policy" | jq -r '.Statement[] | select(.Effect == "Allow")')
+    if [[ -n "$allow_access" ]]; then
+        echo "[OK] Access policy allows necessary access for domain $domain_name."
+    else
+        echo "[FAIL] Access policy does not allow access as required for domain $domain_name." 1>&2
+        SCRIPT_STATUS_OUTPUT=67
+    fi
+}
+
+# Call the function if COMPONENTS_TO_CHECK_IRSA_OS is not empty after excluding components
+if [[ -n "$COMPONENTS_TO_CHECK_IRSA_OS" ]]; then
+    # Use grep -q to check for exclusion
+    if ! echo "$COMPONENTS_TO_CHECK_IRSA_OS" | grep -q -F -x -f <(printf '%s\n' "${EXCLUDE_COMPONENTS_ARRAY[@]}"); then
+        check_irsa_opensearch_requirements
+        check_opensearch_iam_enabled
+    fi
+fi
+
+check_aurora_cluster() {
+    aurora_host="$1"
+    aurora_port="$2"
+    component="$3"
+
+    # Check if the Aurora URL is set
+    if [[ -z "$aurora_host" || "$aurora_host" == "null" ]]; then
+        echo "[FAIL] The Aurora host is not set. Please ensure that $component define it correctly in the Helm chart values." 1>&2
+        SCRIPT_STATUS_OUTPUT=71
+        return
+    fi
+
+    if [[ -z "$aurora_port" || "$aurora_port" == "null" ]]; then
+        echo "[FAIL] The Aurora port is not set. Please ensure that $component define it correctly in the Helm chart values." 1>&2
```

Comment:
same as the one error with the identityKeycloak port check.
Not sure why it's done twice.

Reply:
all good, was just curious whether it's a duplicated check